### PR TITLE
feat: put all linting logic inside lint.sh

### LINKED ISSUE #513: Merge lint check and lint fix logic
**Problem**
Currently we use `./lint.sh` to fix issues locally and on ci. Instead we should use just one script that fixes as well as checks files on CI.

**Solution**
Update `./lint.sh` with the following file. Update CI jobs to use `./lint.sh` instead of manually providing the steps.
```bash
#!/bin/bash

# Configuration for file types to be tested via prettier
FILE_TYPES="{graphql,yml,json,md}"

run_cargo_fmt() {
    MODE=$1
    if [ "$MODE" == "check" ]; then
        cargo +nightly fmt -- --check
    else
        cargo +nightly fmt
    fi
}

run_cargo_clippy() {
    MODE=$1
    CMD="cargo +nightly clippy --all-targets --all-features"
    [ "$MODE" == "check" ] || CMD="$CMD --fix --allow-staged --allow-dirty"
    CMD="$CMD -- -D warnings"
    $CMD
}

run_prettier() {
    MODE=$1
    if [ "$MODE" == "check" ]; then
        prettier --check "**/*.$FILE_TYPES"
    else
        prettier --write "**/*.$FILE_TYPES"
    fi
}

# Extract the mode from the argument
if [[ $1 == "--mode="* ]]; then
    MODE=${1#--mode=}
else
    echo "Please specify a mode with --mode=check or --mode=fix"
    exit 1
fi

# Run commands based on mode
case $MODE in
    check|fix)
        run_cargo_fmt $MODE
        run_cargo_clippy $MODE
        run_prettier $MODE
        ;;
    *)
        echo "Invalid mode. Please use --mode=check or --mode=fix"
        exit 1
        ;;
esac
```

### PR DESCRIPTION
**Pull request by ayewo**
**Summary:**  
_Remove duplication of linting logic in CI by putting it inside `/lint.sh`._

**Issue Reference(s):**  
Fixes #513 

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh` to address and fix linting issues.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [x] I have updated the documentation accordingly (if applicable).
- [x] I have performed a self-review of my own code.

/claim #513

### CODE REVIEW

**Comment by ayewo** [comment]
@tusharmath ready for review..

**Comment by ayewo** [comment]
@tusharmath should I strike out ~I have run ./lint.sh to address and fix linting issues.~ or leave at as checked by default?

**Review by tusharmath — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -65,38 +65,29 @@ jobs:
...
-
+          node-version: "18"
       - name: Install Prettier
         run: npm install --global prettier
-
-      - name: Check Formatting
-        run: prettier --check "**/*.{graphql,yml,json,md}"
-
-      - name: Install Nightly Toolchain
+      - name: Install Stable Toolchain
```
Comment (tusharmath) [inline .github/workflows/ci.yml:74]:
```suggestion
      - name: Install Nightly Toolchain
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -65,38 +65,29 @@ jobs:
...
         uses: actions-rs/toolchain@v1
         with:
           toolchain: nightly
+          override: true
```
Comment (tusharmath) [inline .github/workflows/ci.yml:78]: It wasn't set before ... why do we need to set it to true now?

**Review by tusharmath — COMMENTED** [review]
@ayewo can you look at it again?

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ayewo) [reply .github/workflows/ci.yml:78]: I can change it back. I copy-pasted it from the recently merged changes today: https://github.com/tailcallhq/tailcall/blob/faa72e9ca7d40862ed7b34f45f9fee4650d53713/.github/workflows/bot.yml#L30-L32

**Comment by ayewo** [comment]
@tusharmath resolved all issues in commit [ea0c83a](https://github.com/tailcallhq/tailcall/pull/516/commits/ea0c83a7ffba0ec7dbdcf804ec4217fa5da5ed3e), please take a look.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -65,38 +65,28 @@ jobs:
...
-          toolchain: nightly
-          command: clippy
-          args: --all-targets --all-features -- -D warnings
+          path: |
+            ~/.cargo/registry
+            ~/.cargo/git
+            target
+          key: ${{ runner.os }}-cargo-${{ hashFiles('**/Cargo.lock') }}
+      - uses: actions/checkout@v4
+      - name: Run lint script
```
Comment (tusharmath) [inline .github/workflows/ci.yml:88]:
```suggestion
      - name: Run Lint Script
```

**Review by tusharmath — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ayewo) [reply .github/workflows/ci.yml]: My bad, sorry I missed that. (Not had b/fast yet 😁).

**Pull request merged**